ministryofjustice/laa-apply-for-criminal-legal-aid
# CRIMAP-96 enter case type

## Description of change
Adds the ability to select a case type for a case

## Link to relevant ticket
[CRIMAP-96](https://dsdmoj.atlassian.net/browse/CRIMAP-96)

## Notes for reviewer
- MAAT ids are only optional for Appeal to crown court types (the last two)
- Details on changes are mandatory for Appeals to CC with a change of financial curcumstances
- Please look at the wording of the errors are they clear?

## Screenshots of changes (if applicable)

### After changes:
<img width="1000" alt="Screenshot 2022-08-31 at 16 00 13" src="https://user-images.githubusercontent.com/13377553/187711296-772501b7-8f27-4c62-ba50-75cfafa1af9e.png">

### Drop downs
<img width="791" alt="Screenshot 2022-08-31 at 16 00 48" src="https://user-images.githubusercontent.com/13377553/187711423-ad150f50-a19e-467e-bd11-14bae6e328cd.png">

---

<img width="582" alt="Screenshot 2022-08-31 at 16 01 20" src="https://user-images.githubusercontent.com/13377553/187711569-77f57cd3-17b2-4886-9730-c3b4e4ca0966.png">

## How to manually test the feature
- go through the regular journey until you reach the case type page
- submit blank --> error
- select either appeal type > add MAAT ID > select non appeal type -> error as MAAT ID present
- do the same for last appeal type with the extra details field
- leave MAAT ID blank on an appeals case --> should not error on submission
- leave extra details page blank on Appeal CC + change of financial circs --> should error as field is blank
- if valid should then redirect to codefendants step
